vortex-error/src/lib.rs

```diff
@@ -305,6 +305,14 @@ pub trait VortexExpect {
     /// Returns the value of the result if it is Ok, otherwise panics with the error.
     /// Should be called only in contexts where the error condition represents a bug (programmer error).
     fn vortex_expect(self, msg: &str) -> Self::Output;
+
+    /// Returns the value of the result if it is Ok, otherwise panics with the error.
+    fn vortex_expect_fn(self, msg: impl FnOnce() -> String) -> Self::Output
```

Comment:
Convention is "_with" I think.

But also, you probably shouldn't be putting runtime arguments in the expectation message.

The intention is to explain why the panic isn't possible, not to give debugging info for when it occurs. It's supposed to never occur!

If we really really want this, we should macro it instead, like logging and format macros, which are designed to avoid formatting unless they are used.